tests/test_cargo_build_lib.rs

```diff
@@ -52,3 +53,16 @@ test!(build_with_no_lib {
                 execs().with_status(101)
                        .with_stderr("no library targets found"));
 });
+
+test!(build_with_relative_cargo_home_path {
+    let p = project("foo")
+        .file("Cargo.toml", &basic_bin_manifest("foo"))
+        .file("src/main.rs", r#"
+            fn main() {}
+        "#);
+        env::set_var("CARGO_HOME", "./cargo_home/");
```

Comment:
You'll probably want to set this kind of environment variable just for the subprocess, not for the test process as a whole, but this should end up picking up that change regardless I would think...